SIT.Manager.Avalonia/Services/AkiServerService.cs

```diff
@@ -79,30 +87,51 @@ public class AkiServerService(IBarNotificationService barNotificationService,
                 }
             });
             _process.OutputDataReceived += startedEventHandler;
-            _process.Exited += new EventHandler((sender, e) =>
-            {
+            _process.Exited += new EventHandler((sender, e) => {
                 ExitedEvent(sender, e);
                 IsStarted = false;
             });
 
             _process.Start();
-            if(cal)
-            {
-                _ = Task.Run(() =>
-                {
-                    //This is gross but i genuinely have no other way to do this besides massively overcomplicated systems
-                    System.Threading.Thread.Sleep(10 * 1000);
-                    if(!_process?.HasExited ?? true)
-                    {
-                        IsStarted = true;
-                        ServerStarted?.Invoke(this, new EventArgs());
+            UpdateRunningState(RunningState.Starting);
+
+            if (cal) {
+                _ = Task.Run(async () => {
+                    TarkovRequesting requesting = ActivatorUtilities.CreateInstance<TarkovRequesting>(_serviceProvider, new Uri("http://127.0.0.1:6969/"));
+                    int retryCounter = 0;
+                    while (retryCounter < 6) {
```

Comment:
`retryCounter++`?

Reply:
is at the bottom of the while loop, or is that not what you mean?